src/Analyzer/Passes/LogicalExpressionOptimizerPass.cpp

```diff
@@ -153,9 +153,11 @@ class LogicalExpressionOptimizerVisitor : public InDepthQueryTreeVisitorWithCont
                 }
             };
 
-            if (const auto * lhs_literal = lhs->as<ConstantNode>())
+            if (const auto * lhs_literal = lhs->as<ConstantNode>();
+                lhs_literal && !lhs_literal->getValue().isNull())
```

Comment:
Don't we automatically check for `lhs_literal != nullptr` with assignment `const auto * lhs_literal = lhs->as<ConstantNode>()`?
```suggestion
                !lhs_literal->getValue().isNull())
```

Reply:
We don't, only if you ask for reference (`lhs->as<ConstantNode &>()`) and because I used `if with initilizer`, the initializing part is not part of the check.